magenta/music/musicxml_parser.py

```diff
@@ -465,6 +480,58 @@ def _parse_forward(self, xml_forward):
                * self.state.seconds_per_quarter)
     self.state.time_position += seconds
 
+  def check_time_signature(self):
+    """Correct the time signature for incomplete measures.
+
+    If the measure is incomplete or a pickup, insert an appropriate
+    time signature into this Measure.
+    """
+    # Compute the fractional time signature (duration / divisions)
+    fractional_time_signature = Fraction(self.duration, self.state.divisions)
+
+    if self.state.time_signature is None and self.time_signature is None:
+      # No global time signature yet and no time signature defined
+      # in this measure (no time signature or senza misura).
+      # Insert the fractional time signature as the time signature
+      # for this measure
+
+      # If no time signature is given, assume an underlying quarter note pulse
+      fractional_time_signature = Fraction(self.duration,
+                                           self.state.divisions * 4)
```

Comment:
I checked with Michael Good, and divisions are always parts per quarter note, so the multiplication by four should exist in both places.